python/cudf/dataframe/columnops.py

```diff
@@ -348,11 +353,34 @@ def as_column(arbitrary, nan_as_null=True, dtype=None):
             data = as_column(memoryview(arbitrary))
         except TypeError:
             try:
-                data = as_column(pa.array(arbitrary, from_pandas=nan_as_null),
-                                 dtype=dtype, nan_as_null=nan_as_null)
-            except pa.ArrowInvalid:
-                data = as_column(np.array(arbitrary), dtype=dtype,
-                                 nan_as_null=nan_as_null)
+                pa_type = None
+                if dtype is not None:
+                    if pd.api.types.is_categorical_dtype(dtype):
+                        raise TypeError
+                    else:
+                        np_type = np.dtype(dtype).type
+                        if np_type == np.bool_:
+                            pa_type = pa.bool_()
+                        else:
+                            pa_type = _gdf.np_to_pa_dtype(np.dtype(dtype).type)
+                data = as_column(
+                    pa.array(arbitrary, type=pa_type, from_pandas=nan_as_null),
```

Comment:
Is `from_pandas=nan_as_null` what you intended here?

Reply:
Yes, PyArrow has a parameter `from_pandas` which then converts numpy.nan to nulls so it's the same thing as our parameter.